main.rb

```diff
@@ -0,0 +1,145 @@
+# rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity, Metrics/ModuleLength, Style/For
+module Enumerable
+  # 1.my_each
+  def my_each
+    for i in self
+      yield(i) if block_given?
+    end
+    self
+  end
+
+  # 2. my_each_with_index
+  def my_each_with_index
+    for i in (0..self).size - 1
+      yield(self[i], i) if block_given?
+    end
+    self
+  end
+
+  # 3. my_select
+  def my_select
+    if block_given?
+      result = []
+      my_each do |value|
+        result << value if yield(value)
+      end
+    end
+    result
+  end
+
+  # 4. my_all
+  def my_all?(parameter = nil)
+    result = true
+    if block_given?
+      my_each { |value| return false unless yield(value) }
+    else
+      case parameter
+      when nil
+        my_each { |value| return false if value.nil? || !value }
+      when Regexp
+        my_each { |value| return false unless value =~ parameter }
+      when Class
+        my_each { |value| return false unless value.is_a? parameter }
+      else
+        my_each { |value| return false unless value == parameter }
+      end
+    end
+    result
+  end
+
+  # 4. my_any
+  def my_any?(parameter = nil)
+    result = false
+    if block_given?
+      my_each do |x|
+        result = true if yield(x)
+      end
+      result
+    else
+      case parameter
+      when nil
+        my_each { |x| return true unless x.nil? || !x }
+      when Regexp
+        my_each { |x| return true if x =~ parameter }
+      when Class
+        my_each { |x| return true if x.is_a? parameter }
+      else
+        my_each { |element| result = true unless element.nil? }
+      end
+    end
+    result
+  end
```

Comment:
- #my_any? when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern